### PR TITLE
[PR196 *] Adding documentation for forthcoming changes in 0.4.0

### PR DESCRIPTION
**Pull request by jbohren**
**View online here: http://jbohren-ct.readthedocs.org/en/pre-0.4.0-docs/**

Includes the following:
 * new execution framework design
 * sets local sphinx style to readthedocs style
 * reorganizes main index for easier navigation

To build locally:
```bash
cd docs
make html
```

### Outstanding Issues

- [x] Add documentation on running unit tests (i.e. installing catkin proper) ([comment](https://github.com/catkin/catkin_tools/pull/249#issuecomment-165452329)) (added in b1e4315134abc62e223f3deaa9f32a62e769da8d)
- [x] Add migration guide / comparison guide (#22) (6dae6ce)
- [x] Add documentation for `--mem-limit` (6dae6ce)
- [x] Add documentation for all new features added by #249
 - [x] DESTDIR support
 - [x] update references to build_env.sh
 - [x] document --get-env option
 - [x] document repro lines
 - [x] document extended summary
 - [x] document .catkin file generation
 - [x] document interleaved mode
 - [x] Add "advanced" documentation on the `--env-cache` a.k.a. "turbo" option  (6dae6ce)
- [x] document new features added by #276
 - [x] using linked develspace
 - [x] changes to CLIs
 - [x] develspace layout / architecture
- [x] add troubleshooting note on `-isystem` and `-I` option precedence with overlays
- [x] adds section for external verbs #71

### CODE REVIEW

**Comment by jbohren** [comment]
This has been rebased to incorporate changes from #244 and the new test harness.

**Comment by NikolausDemmel** [comment]
> a.k.a. "turbo" option

This is gold :-D

I've been using it and it really is a turbo for large workspaces.

**Comment by jbohren** [comment]
@wjwwood These docs are all updated for all of the pre-0.4.0 changes and should be good to merge.

File: docs/index.rst [hunk 1]
```diff
@@ -2,84 +2,122 @@ Catkin Command Line Tools
...
+Built-in ``catkin`` Verbs
+-------------------------
 
 Each of the following verbs is built-in to the ``catkin`` command and has its own detailed documentation:
 
 - :doc:`build -- Build packages in a catkin workspace <verbs/catkin_build>`
 - :doc:`config -- Configure a catkin workspace's layout and settings <verbs/catkin_config>`
 - :doc:`clean -- Clean products generated in a catkin workspace <verbs/catkin_clean>`
 - :doc:`create -- Create structures like Catkin packages <verbs/catkin_create>`
+- :doc:`env -- Run commands with a modified environemnt <verbs/catkin_env>`
```
Comment (dirk-thomas) [inline docs/index.rst]: Spelling: environemnt

**Comment by jbohren** [comment]
@wjwwood All tests passing. If you merge this, I can send out the call for testing once the docs are live on rtd.

**Comment by wjwwood** [comment]
@jbohren that sounds good, I'm looking at it now.

**Comment by jbohren** [comment]
@wjwwood Hold on merging while I add a module to sphinx to check spelling, there are more mistakes than the ones @dirk-thomas found, obviously.

**Comment by wjwwood** [comment]
@jbohren great idea. I'm still reviewing it anyways.

**Comment by jbohren** [comment]
@wjwwood 53d1205 should now fail if there are any spelling errors (there still are, but I want to see it break the CI before fixing it)

**Comment by jbohren** [comment]
Ok it looks like it's working on OSX but needs an Ubuntu system package.

**Comment by jbohren** [comment]
We now have CI doc spell checking.

**Comment by wjwwood** [comment]
I rebuilt this branch on RTD, so that I can see the updated output from the shell scripts. Does this need to be rebased since https://github.com/catkin/catkin_tools/pull/293.

**Comment by wjwwood** [comment]
That last sentence should have been a question. Otherwise I think things like the "log" space won't show up in the output of the quickstart page.

**Comment by wjwwood** [comment]
I think you should rebase this before merging. I just looked at the merged result and it resulted in some strange three-way merges which I'm not convinced make sense afterwards. Related to that is that rebasing cannot be done cleanly. I can also not squash it properly. Not sure what's going on.

Also your CI job doesn't work with the latest change (it timed-out because you don't pass `-y` to `apt-get`).

**Comment by wjwwood** [comment]
It's really strange because the commit e5c5224ede4dfc159340293fe99469f9e17d21e2 is in this pr's commit list, but it doesn't show up in the log of a clean `git clone https://github.com/catkin/catkin_tools -b pre-0.4.0-docs`. Not sure what's going on.

**Comment by wjwwood** [comment]
Never mind, I have the commit, but it's way back in the git log. It's a very strange branch.

**Comment by wjwwood** [comment]
@jbohren I pushed a branch of this which is rebased on master: https://github.com/catkin/catkin_tools/tree/pre-0.4.0-docs-rebase (including your latest commit a2e21ee) if you're interested. That one Fix typos commit really screwed things up.

**Comment by wjwwood** [comment]
btw, it looks like the RTD build fails because of your spellchecker: http://readthedocs.org/projects/catkin-tools/builds/3906804/

**Comment by jbohren** [comment]
> @jbohren I pushed a branch of this which is rebased on master: https://github.com/catkin/catkin_tools/tree/pre-0.4.0-docs-rebase (including your latest commit a2e21ee) if you're interested. That one Fix typos commit really screwed things up.

Yeah I created a PR from it as #318. You mean the fix typos commit 472ead0? I was trying to make sure @fudger's commit was in there in a reasonable way, it looks like your rebase did it properly, though.

>btw, it looks like the RTD build fails because of your spellchecker: http://readthedocs.org/projects/catkin-tools/builds/3906804/

I added a separate sphinx conf to be used for the CI only over in #318.

Closing this in favor of #318.

**Pull request closed without merging**